python/taichi/profiler/profiler.py

```diff
@@ -0,0 +1,162 @@
+from taichi.core import ti_core as _ti_core
+from taichi.lang import impl
+
+import taichi as ti
+
+
+class StatisticalResult:
+    def __init__(self, name):
+        self.name = name
+        self.counter = 0
+        self.min_time = 0.0
+        self.max_time = 0.0
+        self.total_time = 0.0
+
+    def __lt__(self, other):
+        if (self.total_time < other.total_time):
+            return True
+        else:
+            return False
+
+    def insert_record(self, time):
+        if self.counter == 0:
+            self.min_time = time
+            self.max_time = time
+        self.counter += 1
+        self.total_time += time
+        self.min_time = min(self.min_time, time)
+        self.max_time = max(self.max_time, time)
+
+
+class Profiler:
+
+    profiling_modes = [True, False]
```

Comment:
is this necessary?

Reply:
profiler test cases may need this 
`for mode in Profiler.profiling_modes:`

Reply:
ohhh these can stay in the doc string (by saying profiling_mode_ is boolean value) so that we can write `for mode in (True, False)`?

Reply:
so far there are only two modes and I have put them in the doc string